classes/core/PKPQueueProvider.php

```diff
@@ -145,6 +147,36 @@ public function boot()
                 ])
             );
         });
+
+        Queue::createPayloadUsing(function(string $connection, string $queue, array $payload) {
+            
+            // If a job dispatch on CLI (e.g. TestJobSuccess/TestJobFailure),
+            // there is simple no way to get the context(journal/press/server) information
+            // and no need to set the context id information in job payload
+            if (runOnCLI()) {
+                return [];
+            }
+
+            return [
+                'context_id' => Application::get()->getRequest()->getContext(),
+            ];
+        });
+
+        Queue::before(function(JobProcessing $event){
+            // Set the context for current CLI session if available right before job start processing
+            // Not necessary when jobs are running via JobRunner as that runs at the end of request life cycle
+            if (runOnCLI()) {
+                Application::get()->setCliContext($event->job->payload()['context_id'] ?? null);
+            }
+        });
```

Comment:
we set the context information to application for each job if available right before job get to process so that following code can get retrieve context information 

```php
Application::get()->getRequest()->getContext();
```